gobblin-compaction/src/main/java/org/apache/gobblin/compaction/CompactorFactory.java

```diff
@@ -29,6 +29,7 @@
 
 /**
  * A factory responsible for creating {@link Compactor}s.
+ * @deprecated
```

Comment:
The annotation on javadoc is intended to explain reasons / alternatives to this API (which you should do). The actual deprecation annotation goes outside of the javadoc (`@Deprecated`)